### PR TITLE
Damage Dealt Utility

### LINKED ISSUE #282: HaveMore/HaveLess matchers
Instead of trying to determine a specific value of damage that is dealt based on some conditions that modify damage (stat stages, held items, moves, abilities, etc.), we can introduce a suite of matchers to check that some value is greater/less than another value. Gomega already supports `BeNumerically(comparator string, compareTo …interface{})`, but we can wrap this value to compare transactions.

The resulting API might look like this:
`Expect(t1).To(HaveMore("Damage", t2))` where `t1`, `t2` are DamageTransactions.

In simpler cases, we can also just check if the targetted Pokemon's HP is less/more after the round, but this won't work for things like accuracy.

### PR DESCRIPTION
**Pull request by adapap**
Adds `DamageDealt([]Transaction, *Pokemon) uint` and `battle.getTarget(int, int) target` helper methods. Started doing a test refactor, would like some input before I continue @dyc3. I am aiming to only change tests where checking for explicit targets is redundant (a.k.a. just comparing damage dealt).

closes #282

### CODE REVIEW

File: util_test.go [hunk 1]
```diff
@@ -343,7 +343,19 @@ func (matcher *orderedTransactionMatcher) NegatedFailureMessage(actual interface
 	)
 }
 
-// Tools for testing the library
+/* Tools for testing the library */
+// Check for damage dealt (if any) by a Pokemon in battle
+func DamageDealt(t []Transaction, p *Pokemon) uint {
```
Comment (dyc3) [inline util_test.go:348]: (nit) Let's make this return an int so we use `Equal(x)` instead of `BeEquivalentTo(x)` in the matchers

**Review by dyc3 — APPROVED** [review]

**Comment by dyc3** [comment]
does this close #282 ?

**Comment by adapap** [comment]
I think so. In my mind I had some idea where you pass in two battles and then compare the damage dealt with a matcher, but I think this is more appropriately handled in a setup/BeforeEach method, or potentially with a `battle.Reset()` method to easily compare two battles.

**Comment by dyc3** [comment]
You could just do a deep copy of the battle to duplicate it

**Pull request merged**